library/src/main/java/org/xmtp/android/library/messages/MessageV2.kt

```diff
@@ -29,6 +29,23 @@ class MessageV2Builder {
             val decrypted =
                 Crypto.decrypt(keyMaterial, message.ciphertext, message.headerBytes.toByteArray())
             val signed = SignedContent.parseFrom(decrypted)
+
+            if (!signed.sender.hasPreKey() || !signed.sender.hasIdentityKey()) {
+                throw XMTPException("missing sender pre-key or identity key")
+            }
+
+            val senderPreKey = PublicKeyBuilder.buildFromSignedPublicKey(signed.sender.preKey)
+            val senderIdentityKey =
+                PublicKeyBuilder.buildFromSignedPublicKey(signed.sender.identityKey)
+
+            if (!senderPreKey.signature.verify(
```

Comment:
Yes if you look at the implementation under the hood with these values it would be

```kotlin
    val ecdsaVerify = java.security.Signature.getInstance("SHA256withECDSA", BouncyCastleProvider())
    ecdsaVerify.initVerify(signed.sender.identityKey.secp256K1Uncompressed.bytes.toByteArray())
    ecdsaVerify.update(signed.sender.preKey.keyBytes.toByteArray())
    return ecdsaVerify.verify(normalizeSignatureForVerification(signed.sender.preKey.signature.rawDataWithNormalizedRecovery))
```